requirements/type-check.txt

```diff
@@ -1,5 +1,6 @@
 -r ci.txt
```

Comment:
probably should add these (and also to the CI cache key)
```suggestion
-r ci.txt
-r unit_tests.txt
-r integration_tests.txt
```

Reply:
Although it's also possible to have conflicts. So maybe type check separately is also viable, i.e. add `type_check_unit_tests` instead of removing `type_check_integration_tests`

Reply:
And when we do it locally, it's always possible to just run them in parallel, i.e. 
```shell
tox run-parallel -e type_check,type_check_unit_tests,type_check_integration_tests
```

Reply:
sg